apps/zotonic_core/src/support/z_context.erl

```diff
@@ -1412,6 +1415,47 @@ set_security_headers(Context) ->
     end,
     cowmachine_req:set_resp_headers(SecurityHeaders1, Context).
 
+flatten_csp(CSP) ->
+    Directives = lists:filtermap(
+        fun
+            ({_Directive, []}) -> false;
+            ({Directive, L}) ->
+                L1 = lists:usort(L),
+                L2 = case lists:member(<<"'unsafe-inline'">>, L1) of
+                    true -> lists:delete(<<"'nonce-'">>, L1);
+                    false -> L1
+                end,
+                D = [ Directive | [ [ " ", V ] || V <- L2 ] ],
+                {true, D}
```

Comment:
Curious: is this necessary or is it merely for convenience?

I suppose `unsafe-inline` makes the `nonce` unnecessary, but is there any downside in keeping both?

Reply:
I remember finding out that if you have both that some browsers start to get confused and apply the stricter nonce, which is a bit weird, but maybe it is hidden somewhere in the specs.